CMakeLists.txt

```diff
@@ -284,39 +308,38 @@ if(ANDROID)
 	# our Qt installation. This is ugly, but it works.
 	set(CMAKE_FIND_ROOT_PATH "/;${CMAKE_FIND_ROOT_PATH}")
 endif()
-set(QT_FIND_COMPONENTS Core Concurrent Widgets Network Svg Positioning Quick Location ${QT_EXTRA_COMPONENTS})
+set(QT_FIND_COMPONENTS Core Concurrent Widgets Network Svg Positioning Quick ${QT_EXTRA_COMPONENTS})
 if (SUBSURFACE_TARGET_EXECUTABLE MATCHES "DesktopExecutable")
-	find_package(Qt5 5.9.1 REQUIRED COMPONENTS ${QT_FIND_COMPONENTS} LinguistTools Test QuickTest)
+	find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS ${QT_FIND_COMPONENTS} LinguistTools Test QuickTest)
```

Comment:
I wonder how Qt Creator is doing it. I've just built it using CMake against Qt 6, but its CMakeLists.txt files still say "Qt5". 

See https://github.com/qt-creator/qt-creator/blob/master/CMakeLists.txt#L56-L68

Reply:
so according to the docs there are three different ways you can do this. One works only with Qt 5.15 and newer. One works only up to Qt 5.15. And the way I did it is the one that's recommended for Qt 5 or Qt 6.
It's entirely possible that I simply followed bad instructions... but in my defense, what I have works :)

Reply:
Well, I'm not a CMake expert either. But it's possible there were some changes in 5.15 to allow for 5 & 6 simplification, but only if you can require 5.15 on the first place and it looks like you can't.